core/trino-main/src/main/java/io/trino/connector/system/SystemConnectorModule.java

```diff
@@ -79,6 +81,8 @@ public void configure(Binder binder)
 
         binder.bind(GlobalSystemConnectorFactory.class).in(Scopes.SINGLETON);
         binder.bind(SystemConnectorRegistrar.class).asEagerSingleton();
+        newOptionalBinder(binder, MaterializedViewsSystemTableAdapter.class)
```

Comment:
We already added tests in the 2nd commit which won't pass without the changes in this commit.
This commit should be squashed as well.

Reply:
Commit squashed